src/numericalAnalysis.cpp

```diff
@@ -0,0 +1,119 @@
+#include "mantella_bits/numericalAnalysis.hpp"
+
+// C++ standard library
+#include <cmath>
+#include <limits>
+
+// Mantella
+#include "mantella_bits/assert.hpp"
+
+namespace mant {
+  double brent(
+      std::function<double(double)> objectiveFunction,
+      double lowerBound,
+      double upperBound,
+      double xTolerance) {
+    double lowerBoundObjectiveValue = objectiveFunction(lowerBound);
+    double upperBoundObjectiveValue = objectiveFunction(upperBound);
+
+    verify(lowerBoundObjectiveValue * upperBoundObjectiveValue <= 0.0, "brent: The lower bound and upper bound objective value must have a different sign.");
```

Comment:
After reading this line (especially the equality part) I was questioning myself: How long will it take for us to recognise if any bound is the root or at least close enough? Maybe we should add something like:

```
if (lowerBoundObjectiveValue < 1e-12) {
  // The lower bound is already close enough to the root.
  return lowerBound;
}
if (upperBoundObjectiveValue < 1e-12) {
  // The upper bound is already close enough to the root.
  return upperBound;
}
```